From 1fd1a2acd6d9e9811abcf661c226abc7c01e0c05 Mon Sep 17 00:00:00 2001
Message-ID: <1fd1a2acd6d9e9811abcf661c226abc7c01e0c05.1700222319.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1700222319.git.sharlatanus@gmail.com>
References: <cover.1700222319.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Wed, 15 Nov 2023 17:52:14 +0000
Subject: [PATCH 24/32] gnu: python-poliastro: Improve package style.

* gnu/package/astronomy.scm (python-poliastro): Improve package style.
[arguments]: Swap to use {test-flags} and standard {check} phase.

Change-Id: I0dde01af97adc883e8a3ce82a8448f4d10fdab64
---
 gnu/packages/astronomy.scm | 27 +++++++++++----------------
 1 file changed, 11 insertions(+), 16 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index d12dccc208..f19f40a69d 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -2118,27 +2118,22 @@ (define-public python-poliastro
     (build-system pyproject-build-system)
     (arguments
      (list
+      #:test-flags
+      #~(list "-m" "not remote_data"
+              ;; TODO: Review failing tests later when any upstream
+              ;; suggestions are provided:
+              ;; https://github.com/poliastro/poliastro/issues/1618
+              "-k" (string-append "not test_czml_add_trajectory"
+                                  " and not test_czml_custom_packet"
+                                  " and not test_czml_ground_station"
+                                  " and not test_czml_groundtrack"
+                                  " and not test_czml_preamble"))
       #:phases
       #~(modify-phases %standard-phases
           ;; NOTE: Tests take about 7-10 minutes to pass.
           (add-before 'check 'prepare-test-environment
             (lambda _
-              (setenv "HOME" "/tmp")
-              ;; TODO: Review failing tests later when any upstream
-              ;; suggestions are provided:
-              ;; https://github.com/poliastro/poliastro/issues/1618
-              (substitute* "tests/test_czml.py"
-              (("def test_czml_add_trajectory") "def __off_test_czml_add_trajectory")
-              (("def test_czml_custom_packet") "def __off_test_czml_custom_packet")
-              (("def test_czml_ground_station") "def __off_test_czml_ground_station")
-              (("def test_czml_groundtrack") "def __off_test_czml_groundtrack")
-              (("def test_czml_preamble") "def __off_test_czml_preamble"))))
-          (replace 'check
-            (lambda* (#:key tests? #:allow-other-keys)
-              (when tests?
-                (invoke "python" "-m" "pytest"
-                        ;; Skip tests that need remote data.
-                        "-m" "not remote_data")))))))
+              (setenv "HOME" "/tmp"))))))
     (native-inputs
      (list python-coverage
            python-hypothesis
-- 
2.41.0

